plugins/generator-1.20.4/neoforge-1.20.4/templates/elementinits/entities.java.ftl

```diff
@@ -75,9 +75,11 @@ public class ${JavaModName}Entities {
 
 	<#if entitiesWithInventory?size != 0>
 	<#compress>
-	@SubscribeEvent public static void registerCapabilities(RegisterCapabilitiesEvent event) {
+	<#-- #4780: entities have a "vanilla" inventory capability registered before custom ones without priority set -->
+	@SubscribeEvent(priority = EventPriority.HIGHEST) public static void registerCapabilities(RegisterCapabilitiesEvent event) {
 		<#list entitiesWithInventory as entity>
-			event.registerEntity(Capabilities.ItemHandler.ENTITY, ${entity.getModElement().getRegistryNameUpper()}.get(), (living, context) -> living.getInventory());
+			event.registerEntity(Capabilities.ItemHandler.ENTITY, ${entity.getModElement().getRegistryNameUpper()}.get(),
+				(living, context) -> ((${entity.getModElement().getName()}Entity) living).getCombinedInventory());

```

Comment:
Due to how types are handled in registerEntity, cast `(${entity.getModElement().getName()}Entity)` is not really needed so you can remove it and revert to one-line

Reply:
Last time I tried I got method absence error in the console.

Reply:
:heavy_check_mark: Fixed